systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeST.java

```diff
@@ -0,0 +1,212 @@
+
+                // sort keys and pick 'next version'
+                SortedSet<String> sortedKeys = new TreeSet<>(mapOfKafkaVersionsWithSupportedClusterOperators.keySet());
+                Iterator<String> kafkaVersions = sortedKeys.iterator();
+                String[] newKafkaVersion = new String[1];
+
+                while (kafkaVersions.hasNext()) {
+                    String kafkaVersion = kafkaVersions.next();
+                    if (kafkaVersion.equals(currentKafkaVersion)) {
+                        LOGGER.info("This is current version {} but we need next one!", kafkaVersion);
+                        if (kafkaVersions.hasNext()) {
+                            newKafkaVersion[0] = kafkaVersions.next();
+                            LOGGER.info("New Kafka version is {} and we are gonna update Kafka custom resource.", newKafkaVersion[0]);
+                        }
+                    }
+                }
+
+                if (newKafkaVersion[0] == null || newKafkaVersion[0].isEmpty()) {
+                    throw new RuntimeException("There is not new Kafka version! Latest is:" + currentKafkaVersion);
+                }
+
+                Map<String, String> kafkaSnapshot = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+                // we are gonna use latest Kafka
+                if (containerImageTag.equals("6.6.6")) {
+                    newKafkaVersion[0] = sortedKeys.last();
+                }
+
+                KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+                    //  2.2.1 -> 2.2 (gonna trim from kafka version)
+                    String logMessageFormatVersion = newKafkaVersion[0].substring(0, 2);
+                    LOGGER.info("We are gonna update Kafka CR with following versions:\n" +
+                        "Kafka version: {}\n" +
+                        "Log message format version: {}", newKafkaVersion[0], logMessageFormatVersion);
+                    kafka.getSpec().getKafka().getConfig().put("log.message.format.version", logMessageFormatVersion);
+                    kafka.getSpec().getKafka().setVersion(newKafkaVersion[0]);
+                });
+
+                // wait until RU
+                StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaSnapshot);
+            }
+        }
+
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBridgeExampleClients.Builder()
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(CLUSTER_NAME))
+            .withTopicName(topicUpgradeName)
+            .withMessageCount(messageUpgradeCount)
+            .withDelayMs(1)
+            .build();
+
+        kafkaBasicClientJob.producerStrimzi().done();
+        kafkaBasicClientJob.consumerStrimzi().done();
```

Comment:
You should attach clients at the beginning of the upgrade as we do it in StrimziUpgradeST.